DPPP/H5ParmPredict.cc

```diff
@@ -110,19 +109,18 @@ namespace DP3 {
       info().setNeedVisData();
       info().setWriteData();
 
-      vector<Predict::ShPtr>::iterator predictstep;
-      for (predictstep=itsPredictSteps.begin();
-           predictstep!=itsPredictSteps.end();
-           predictstep++) {
-        (*predictstep)->updateInfo(infoIn);
+      for(Predict::ShPtr& predictstep : itsPredictSteps)
+      {
+        predictstep->setNThreads(NThreads());
```

Comment:
This is the actual fix :)

Reply:
Would it be more useful to embed nthreads in DPInfo? That way you cannot forget to set it (it's set to a default in DPRun I think).

Reply:
I considered it, but I thought there was some value in having a separation of things which come out of the MS and program options -- e.g. the parset is also not part of DPInfo. Separately setting nthreads was a bit more of a hassle than I thought though, so whether it was the right choice, hmm. We can reconsider if we change things related to it.